---
title: "GS in CIMMYT Wheat"
author: "Make this your own"
date: "`r format(Sys.time(), '%B %d, %Y')`"
output: html_document
editor_options: 
  chunk_output_type: console
---
#Objective

Set up a genomic selection for CIMMYT wheat trials.  The data included are a hapmap file that has previously been imputed using Beagle.  The phenotype data file consist of 3 site years of data, 14_OBR_EYT--2014 Drought, 15_OBR_EYT -- 2015 Drought, 14_OBR_EYTHeat -- 2015 heat.  Note that not all phenotypes were included each year.  The lines were grouped into trials, with trials composing sister (highly related lines).

The phenotype information includes:  
GID_DNA:  This is the individual unique number that links to the hapmap file
blup_gs1:  Model estimated value for NDVI 1st measurement 	
blup_gs2:	Model estimated value for NDVI 2nd measurement
blup_gs3: Model estimated value for NDVI 3rd measurement	
blup_n1:	Model estimated value for canopy temperature 1st measurement
blup_n2:	Model estimated value for canopy temperature 2nd measurement
blup_n3:	Model estimated value for canopy temperature 3rd measurement
Experiment_id:  Identifies year and type of experiment heat or drought	
Entry:  Entry number	
cimmyt_tid:  CIMMYT Trial ID, grouping of individuals into trial	
notes:  notes	
gy_effect:  Estimated BLUP (0 centered) of grain yield	
gs: average of blup_gs1, blup_gs2, blup_gs3, and blup_gs4	
ir: average of blup_n1, blup_n2, blup_n3, blup_n4	
model: information about the model used to calcuate trial BLUPs	
blup_gs4:	Model estimated value for NDVI 4th measurement
blup_n4: Model estimated value for canopy temperature 4th measurement

This data is from Crain et al. (2018), Combining High-Throughput Phenotyping and
Genomic Information to Increase Prediction and Selection Accuracy in Wheat Breeding.  The data sets and scripts are stored at https://datadryad.org/resource/doi:10.5061/dryad.7f138.

#Objective 1:  Set up a model using only the drought 2014 data for grain yield.

```{r setup, eval=TRUE}
knitr::opts_knit$set(root.dir = '')
setwd('')
options(digits = 10) #set options to include lots of digits
#Git link
#https://github.com/jcrain1/BecA
```

```{r get_functions}
####Function to count alleles and populations parameters###
tassel5_to_params=function(x="hap matrix", y="columns to skip", z="population number"){
      geno=x
      #recount allele A and B and het
      alleleA=rowSums(geno[,(y+1):ncol(geno)]!=substring(geno$alleles, 3, 3) & geno[,(y+1):ncol(geno)]!="N") #only counts what is not allele B and missing.  i.e. counts allele A and various calls for heterozygous
      alleleB=rowSums(geno[,(y+1):ncol(geno)]!=substring(geno$alleles, 1, 1) & geno[,(y+1):ncol(geno)]!="N")
      het=rowSums(geno[,(y+1):ncol(geno)] == "M") + rowSums( geno[,(y+1):ncol(geno)] ==   "R") + rowSums(geno[,(y+1):ncol(geno)] ==  "W") + rowSums(geno[,(y+1):ncol(geno)] ==  "K") + rowSums(geno[,(y+1):ncol(geno)] ==  "S") + rowSums(geno[,(y+1):ncol(geno)] ==  "Y")
      present=1-(rowSums(geno[,(y+1):ncol(geno)]=="N")/z)
      MAF=apply(cbind(((alleleA-het)*2+het), (alleleB-het)*2+het), 1, min)/apply(
    cbind(((alleleA-het)*2+het), ((alleleB-het)*2+het)), 1, sum) 
      percentHet=het/apply(cbind(alleleA-het, alleleB-het, het), 1, sum)
      return(cbind.data.frame(geno[,1:y], "alleleA"=alleleA, "alleleB"=alleleB, "het"=het, "present"= present, "MAF"=MAF, "percentHET"=percentHet, geno[,(y+1):ncol(geno)]))
}

##function to convert hap to 0 and 1
hap_to_G=function(x="hap matrix", y="number of columns of information"){
  ##From Prasana, pulls out first allele for a and second for b
  a = substring(x$alleles,1,1)
  #Checks the frequency of the alleles if the second allele is more frequent it is substitued
  a[x$alleleA<x$alleleB] = substring(x$alleles,3,3)[x$alleleA<x$alleleB]
  #Same thing with the second allele
  b = substring(x$alleles,3,3)
  b[x$alleleA<x$alleleB] = substring(x$alleles,1,1)[x$alleleA<x$alleleB]
  #Checks to make sure all alleles are one or the other
  #print(paste("If 0 all alleles are accounted for: ", sum(a == b), sep=""))
  
  ## Turn into letter matrix for mapping
  #makes a copy of the hap matrix
  hap01 = x
  #sets all allele values to NA
  hap01[,(y+1):ncol(hap01)]=NA
  
  ## Turn allele a and allele b into 1 and -1.  Het into 0
  #line by line if a line is a then it places 1 in hap01 for the allele
  hap01[x == a] = 1
  hap01[x == b] = -1
  hap01[x == "M"] = 0
  hap01[x == "Y"] = 0
  hap01[x == "K"] = 0
  hap01[x == "R"] = 0
  hap01[x == "W"] = 0
  hap01[x == "S"] = 0
  hap01[x== "N"]=NA
  
  return(hap01)}
```

#Load Data

```{r load_data}


```

#Examine Data
```{r examine_data}
#potential genotype filtering
#potential phenotype filtering
```

#Set up Model
```{r gs_model}

```

#Validate Model
```{r examine_model_accuracy}


```

#Objective 2 Make your own model for a trait and experiment

#R Session Information
```{r what_packages_and_version}

```